src/util/misc/projectStroke.ts

```diff
@@ -39,62 +49,216 @@ export const projectStrokeOnPoints = (
-      scalar = -s / Math.sin(alpha / 2);
-      miterVector = new Point(
-        bisectorVector.x * scalar * strokeUniformScalar.x,
-        bisectorVector.y * scalar * strokeUniformScalar.y
-      );
-      if (Math.hypot(miterVector.x, miterVector.y) / s <= strokeMiterLimit) {
-        coords.push(A.add(miterVector));
-        coords.push(A.subtract(miterVector));
+      alpha = Math.abs(bisector.angle); //TODO: check if ABS is really needed. I don't remember why I implemented it this way
+
+    /* MITER 
+      Calculation: the corner is formed by extending the outer edges of the stroke at the tangents of the path segments until they intersect
+      Visually detailed here: https://github.com/fabricjs/fabric.js/issues/8025
+    */
+    if (strokeLineJoin === 'miter') {
+      const scalar = -s / Math.sin(alpha / 2),
+        miterVector = scaleHatVector(bisectorVector, scalar);
+
+      // When two line segments meet at a sharp angle, it is possible for the join to extend,
+      // it is possible for the join to extend far beyond the thickness of the line stroking
+      // the path. The stroke-miterlimit imposes a limit on the extent of the line join.
+      // MDN: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit
+      // When the stroke is uniform, scaling changes the arrangement of points, this changes the miter-limit
+      let adjustedStrokeMiterLimit;
+      if (strokeUniform) {
+        // TODO: As it is in the MDN, the miterLimit can also be calculated as 1/sin(alpha/2) , I believe it is more performant than the current implementation.
+        const miterLimitVector = scaleHatVector(bisectorVector, strokeMiterLimit * s);
+        adjustedStrokeMiterLimit = hypot(miterLimitVector.x, miterLimitVector.y) / s;
+      } else {
+        adjustedStrokeMiterLimit = strokeMiterLimit
+      }
+
+      if (hypot(miterVector.x, miterVector.y) / s <= adjustedStrokeMiterLimit) {
+        const proj1 = applySkew(A.add(miterVector));
+        coords.push({
+          "projectedPoint": proj1,
+          "originPoint": A,
+          "bisector": bisector
+        });
         return;
       }
     }
-    scalar = -s * Math.SQRT2;
-    miterVector = new Point(
-      bisectorVector.x * scalar * strokeUniformScalar.x,
-      bisectorVector.y * scalar * strokeUniformScalar.y
-    );
-    coords.push(A.add(miterVector));
-    coords.push(A.subtract(miterVector));
+    /* ROUND (without skew)
+      Calculation: the projections are the two vectors parallel to X and Y axes
+      Visually detailed here: https://github.com/fabricjs/fabric.js/pull/8083
+    */
+    if (strokeLineJoin === 'round' && skewX === 0 && skewY === 0) {
+      // correctSide is used to only consider projecting for the outer side 
+      const correctSide = new Point(
+          Math.abs(Math.atan2(bisectorVector.y, bisectorVector.x)) >= PiBy2 ? 1 : -1, 
+          Math.abs(Math.atan2(bisectorVector.x, bisectorVector.y)) >= PiBy2 ? 1 : -1
```

Comment:
utilize `calcAngleBetweenVectors`?

Reply:
It's a little different here, I'm not calculating the angle between vectors.. I'm just checking which quadrant the bisector belongs to to find out which of the two projections will be chosen.

Reply:
Well you are... between the vector and the x hat vector

Reply:
Ok, you're right.. It seems to me to be more intelligible, although I have to instantiate a class

```
Math.abs(calcAngleBetweenVectors(new fabric.Point(1, 0), bisectorVector))
```
..than this:

```
Math.abs(Math.atan2(bisectorVector.y, bisectorVector.x))
```